apis/v1/gatewayclass_types.go

```diff
@@ -261,9 +261,10 @@ type GatewayClassStatus struct {
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 
```

Comment:
Istio basically messed up by not having a feature flag for the old setting (like Envoy Gateway). This is the only field we have this issue on since its the only experimental write (not read) field.

So if this is merged, all users who upgrade to a newer GW API are going to get ~10 pages of error logs in Istio. I can understand its experimental and that may not factor in much, but just saying.

Reply:
I was about to leave a comment to that extent, asking about failure modes of existing controllers here. Does the failure become any less painful if we rename/remove the field as part of this transition?

Reply:
just to learn more, what do you mean old setting? the supportedFeatures API in its current form?

Reply:
I agree that it sucks that a breaking change will produce a bunch of errors, but it's the risk we all run when supporting experimental features.

Renaming the field would make things easier, but I can't think of a name that makes sense.